odl/solvers/util/callback.py

```diff
@@ -1035,6 +1036,82 @@ def __repr__(self):
                                    inner_str)
 
 
+@contextlib.contextmanager
+def save_animation(filename,
+                   writer="ffmpeg",
+                   writer_kwargs=None,
+                   dpi=200,
+                   saving_kwargs=None,
+                   fig=None,
+                   step=1):
+    """Context manager for creating animations using `matplotlib.animation` [1]
+    [1] https://matplotlib.org/3.2.0/api/animation_api.html
+
+    Parameters
+    ----------
+    filename : str
+    writer : string
+        Available writers: `matplotlib.animation.writers.list()`, examples:
+        'ffmpeg' -> .mp4 file, 'pillow' -> .gif file
+    writer_kwargs : dict
+        Containing e.g. `fps` and a dict of `metadata` like
+        `title`, `author`, ...
+        Passed to `matplotlib.animation.writer`.
+    saving_kwargs : dict
+        Containing e.g. `fps` and a dict of `metadata` that may contain
+        title, artist, genre, subject, copyright, srcform, comment.
+        Passed to `matplotlib.animation.writer.saving`.
+    fig : matplotlib figure, optional
```

Comment:
```suggestion
    fig : matplotlib.figure.Figure, optional
```
This might be linked correctly by Sphinx.